common/persistence/sql/sqlplugin/mysql/events.go

```diff
@@ -63,10 +63,10 @@ const (
 	getHistoryTreeQuery = `SELECT branch_id, data, data_encoding FROM history_tree WHERE shard_id = ? AND tree_id = ? `
 
 	paginateBranchesQuery = `SELECT shard_id, tree_id, branch_id, data, data_encoding
-        FROM history_tree
-        WHERE (shard_id, tree_id, branch_id) > (?, ?, ?)
-        ORDER BY shard_id, tree_id, branch_id
-        LIMIT ?`
+		FROM history_tree
+		WHERE shard_id = ? AND ((tree_id = ? AND branch_id > ?) OR tree_id > ?) OR shard_id > ?
```

Comment:
maybe add some simple comment here that we tried the tuple comparison but it result in mysql doing stupid thing